### ISSUE TITLE
issues with html email

### ISSUE THREAD

**Issue by jvanasco** [Pylons/pyramid_mailer#42]
I kept on getting this error on generating an email 

```
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2019' in position 1122: ordinal not in range(128)
```

issues seem to sit in the standard lib's email.message module and the `Message.as_string` 

digging into the pyramid_mailer and repoze.sendmail projects, I'm pretty sure this is what is happening:
- the html email component ( that is dropped in through the html kwarg ) is kept as raw text and not encoded into base64
- when i try to pipe the result of `msg.to_message` into a process ( via popen, etc ), it's just the **str** representation , which points to Message.as_string function

I need to somehow get an ascii_encoded (b64) version of the Message object.  

Any ideas on acceptable ways to handle this ?

My thoughts so far are to either b64 encode within pyramid_mailer.response when needed ( eg, if you set html/text , we quickly try to encode to ascii - if any issues arise, it is encoded as an attachment ) OR figure out what is going on in the actual attachment phase of MailResponse.to_message , because it should be encoded at that point.

basically, i need this test to pass:

```
def test_attach_as_body_and_html__ascii_safe(self):

    from pyramid_mailer.mailer import Mailer
    from pyramid_mailer.message import Message

    mailer = Mailer()

    utf_8_encoded = b('mo \xe2\x82\xac')
    utf_8 = utf_8_encoded.decode('utf_8')

    text_string = utf_8
    html_string = '<p>' + utf_8 + '</p>'

    msg = Message(subject="testing",
                  sender="sender@example.com",
                  recipients=["tester@example.com"],
                  body=text_string,
                  html=html_string)

    message = msg.to_message()

    # test a        
    as_ascii = str(message)

    # test b
    as_ascii = message.to_ascii()
```
